pkg/compiler/admin.go

```diff
@@ -0,0 +1,37 @@
+package compiler
+
+import (
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/lyft/flyteidl/gen/pb-go/flyteidl/core"
+)
+
+// Should we call this something else? It can be used more generally if ever necessary.
```

Comment:
Historically: I wanted to make both tasks and launchplan  parameters in the compiler to take InterfaceProviders... but then launch plans diverged  (with expected* stuff) maybe there is still hope?

Reply:
I can't think of a good name... InterfaceProviderProvider was the best I could come up with, so I'm leaving it as is for now.